### PR TITLE
Add dvm web3 test

### PR DESCRIPTION
**Pull request by AsceticBear**
This PR contains a bunch of test scripts about the DVM RPC interface. I attached these tests to CI/CD to help us find potential problems early when making changes.

The CI workflow now:

Run Test                             ---> cargo test
Run Run DVM RPC Test    ---> cargo build + dvm rpc tests

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -28,3 +28,30 @@ jobs:
 
       - name: Overall test
         run: cargo test
+      
+  dvm-rpc-test:
+    if: "contains(toJSON(github.event.head_commit.message), '[skip dvm]')"
```
Comment (wi1dcard) [inline .github/workflows/ci.yml:33]: Should this be something like `!contains(..., '[skip dvm]')`?

File: .gitignore [hunk 2]
```diff
@@ -21,3 +21,6 @@ release/
 .cargo-ok
 Cargo.toml.bak
 expand.rs
+
+# Js
+**/node_modules/
```
Comment (wi1dcard) [inline .gitignore:26]: It could (not 100% sure) work without the leading `**/`

Comment (AsceticBear) [reply .gitignore:26]: Indeed, it works well without `**/`.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (AsceticBear) [reply .github/workflows/ci.yml]: A bit weird here. The second action workflow will be triggered when `if: "!contains(toJSON(github.event.head_commit.message), '[skip dvm]')"`. But, without `!`, it works well.

Comment (AsceticBear) [reply .github/workflows/ci.yml]: I have tried hundred times with the head_commit flag to skip the DVM test for specific situations. Just ignore this feature at present.

**Review by HackFisher — APPROVED** [review]

**Comment by HackFisher** [comment]
"Build and Test Expected — Waiting for status to be reported"

@wi1dcard why github report this which prevent merging?

**Comment by wi1dcard** [comment]
> "Build and Test Expected — Waiting for status to be reported"
> 
> @wi1dcard why github report this which prevent merging?

Fixed. We set protected branches and "required status checks to pass before merging" in the repo settings, after renaming the workflow or job name, we'd have to change corresponding settings.

image

**Pull request merged**